code/src/components/Shop.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { shoppings } from '../reducers/shoppings'
+import { Button } from '../styledComponents/Button'
+import { DoubleButtons } from '../styledComponents/DoubleButtons'
+import { List } from '../styledComponents/List'
+import { ListItem } from '../styledComponents/ListItem'
+import { CustomCheckBox } from 'components/CustomCheckBox'  
+
+
+export const Shop = ({itemIndex}) => {
+  const item = useSelector(store => store.shopping.list.items[itemIndex])
+  const dispatch = useDispatch()
+
+  const handleOnChange = event => {
+    dispatch(shoppings.actions.setDoneShop({
+      itemIndex: itemIndex,
```

Comment:
when key and value is the same, you can put it together in one word.
so instead of `itemIndex: itemIndex,` your can put just `itemIndex,`

HEads up though! The way your code is setup the index changes on your todos, which is totally fine. But that makes a checked todo when it removed, be checked on the one below... Hard to explain in text but I'm adding a little screen recording of it.
Try to use an id instead of an index when changing and removing your todos. Many have made this same mistake. It can work with index as well (Since Van was showing you that) But it depends on how you deal with the global state..